Marlin/src/module/planner.cpp

```diff
@@ -2122,13 +2122,15 @@ bool Planner::_populate_block(block_t * const block, bool split_move,
 
       #if ENABLED(DISABLE_INACTIVE_EXTRUDER) // Enable only the selected extruder
 
-        LOOP_L_N(i, EXTRUDERS)
+        LOOP_L_N(i, E_STEPPERS)
           if (g_uc_extruder_last_move[i]) g_uc_extruder_last_move[i]--;
 
+        #define E_STEPPER_INDEX(E) TERN(SWITCHING_EXTRUDER, (E) / 2, E)
+
         #define ENABLE_ONE_E(N) do{ \
           if (extruder == N) { \
```

Comment:
Shouldn be this /2? N is looping with E_STEPPER

Reply:
I was writting a define for dividing 'extruder' but surely I have not understood logic

Reply:
something like:
`#define GET_EXTRUDER_STEPPER(E) TERN(SWITCHING_EXTRUDER, (E) >> 1, E)`
`if (GET_EXTRUDER_STEPPER(extruder( == N) { \`

Reply:
Yes, that is correct. Patched.